pkgs/development/python-modules/sqlcipher3/default.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  sqlcipher,
+  openssl,
+}:
+let
+  pname = "sqlcipher3";
+  version = "0.5.4";
+in
+buildPythonPackage {
+  inherit pname version;
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-4w/1jWTdQ+Gezt3RARahonrR2YiMxCRcdfK9qbA4Tnc=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  buildInputs = [
+    sqlcipher
+    openssl
+  ];
+
+  meta = with lib; {
+    mainProgram = "sqlcipher3";
+    homepage = "https://github.com/coleifer/sqlcipher3";
+    description = "Python 3 bindings for SQLCipher";
+    platforms = platforms.unix;
```

Comment:
```suggestion
```
`buildPythonPackage` sets `meta.platforms` by default.

Reply:
@phaer that was your code, would you mind to edit this? (Push to belamu:353082-bump-signal-export-to-3.2.0 should be possible for you)

My question again (from a learning ambition): where could I've seen this information? The only documentation of buildPythonPackage I've found was [here](https://dream2nix.dev/reference/buildPythonPackage/) which did not write about meta(.platforms). So far I haven't l found/learned where buildPythonPackage (or similar) are defined to see this information in source code.

Reply:
It is the dream2nix documentation, not the nixpkgs one.
The nixpkgs documentation is [here](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck). (or [rendered one](https://nixos.org/manual/nixpkgs/unstable/#python))

Unfortunately, `meta` is not mentioned, but the implementation is here.
https://github.com/NixOS/nixpkgs/blob/f93401eeed99c853c8c92158555dcc17eae4d914/pkgs/development/interpreters/python/mk-python-derivation.nix#L326-L330

Reply:
> @phaer that was your code, would you mind to edit this? (Push to belamu:353082-bump-signal-export-to-3.2.0 should be possible for you)

Sorry, I've been busy and the invitation expired. Can you try adding the test hook and removing this line, before we go back and forth on this again? :)

Reply:
@phaer Sorry, I was not able to. Removing the meta line of course but the tests are more confusing for me.
What I figured out:
- tests for sqlcipher3 are supposed to be run [by calling `python tests/`](https://github.com/coleifer/sqlcipher3/blob/e2176211155d0cc0a6580a9d9788b84fd77ef254/.github/workflows/tests.yaml#L24C14-L24C27)
- the tests use `unittest` which is supposed to be [compatible](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-unittestcheckhook-using-unittestcheckhook) with `pytest`
- adding `nativeCheckInputs= [ pytestCheckHook ]; pytestFlagsArray = [ "tests/" ];` (and `pytestCheckHook` in the arguments at the top of `default.nix`) results in error because `tests` is not found
- the tar that is downloaded (in /nix/store/8232cdbi6vxb55l1hdvkk6f7xrf3p8np-sqlcipher3-0.5.4.tar.gz ) does not contain a directory `tests`
  - do we need to download a complete source?
- `pythonImportsCheck` seems to be needed for python packages that are imported. The imported modules are `datetime`, `time`, `collections.abc`. But to check them, they would need to be listed in the arguments at the top of `default.nix` I guess which seems unreasonable for standard modules. Not sure if the needed non-python-dependencies should be checked with `pythonImportsCheck`.

If you understand that better, I would be glad if you could fix this.

Reply:
Hello!

I just (force-)pushed to your branch.

Just added pythonImportChecks hook. That one checks that the exported module, sqliteciphers3 in this case, can be imported - which gives some indication that the build is not completely broken.

The problem with pytestCheckHook here is that the test files don't start with "test_". We would need to patch in a pytest.ini or so with "python_files" set, but then we run into a nixpkgs problem where the built python extension is only installed in the output, not in the build directory where the checkPhase runs by default.

We could fix that, but I'd prefer to do that in another PR if needed. You are also correct that we would have to download the source from github or so for that to work.